### PR TITLE
docs: fix incorrect information about webp support in the source prop of React Native's Image component

### PR DESCRIPTION
**Pull request by muhammadashfaq**
**Summary:**
This PR corrects the misinformation regarding `webp` image format support in the `source` prop of React Native's `Image` component documentation. The current description incorrectly states that `webp` is only available when bundled with JavaScript on iOS, which is inaccurate. 

**Changes:**
- Updated the `Image` component's `source` prop description to accurately reflect the platform support for `webp` format.
- Clarified that `webp` is supported on Android by default and on iOS when using libraries like `react-native-webp` or `SDWebImage`.

**Impact:**
This correction ensures that developers are provided with accurate information regarding image format support across platforms, helping avoid confusion and potential implementation errors.

### CODE REVIEW

**Comment by muhammadashfaq** [comment]
@cortinico

File: docs/image.md [hunk 1]
```diff
@@ -419,7 +419,7 @@ The image source (either a remote URL or a local file resource).
 
 This prop can also contain several remote URLs, specified together with their width and height and potentially with scale/other URI arguments. The native side will then choose the best `uri` to display based on the measured size of the image container. A `cache` property can be added to control how networked request interacts with the local cache. (For more information see [Cache Control for Images](images#cache-control-ios-only)).
 
-The currently supported formats are `png`, `jpg`, `jpeg`, `bmp`, `gif`, `webp`, `psd` (iOS only). In addition, iOS supports several RAW image formats. Refer to Apple's documentation for the current list of supported camera models (for iOS 12, see https://support.apple.com/en-ca/HT208967).
+The currently supported formats are `png`, `jpg`, `jpeg`, `bmp`, `gif`,`webp` (Android only), `psd` (iOS only). In addition, iOS supports several RAW image formats.`webp` format is supported on iOS only when bundled with the JavaScript code. Refer to Apple's documentation for the current list of supported camera models (for iOS 12, see https://support.apple.com/en-ca/HT208967).
```
Comment (cortinico) [inline docs/image.md:422]: > `webp` format is supported on iOS only when bundled with the JavaScript code

Where is this coming from? Do you have a code reference from React Native or a GitHub issue for reference?

Comment (muhammadashfaq) [reply docs/image.md:422]: @cortinico 
Github Issue is opened.
https://github.com/facebook/react-native-website/issues/4165

Comment (cortinico) [reply docs/image.md:422]:
```suggestion
The currently supported formats are `png`, `jpg`, `jpeg`, `bmp`, `gif`,`webp`, `psd` (iOS only). In addition, iOS supports several RAW image formats. Refer to Apple's documentation for the current list of supported camera models (for iOS 12, see https://support.apple.com/en-ca/HT208967).
```

Comment (cortinico) [reply docs/image.md:422]: And let's add a separate paragraph about webp below please

Comment (muhammadashfaq) [reply docs/image.md:422]: Sure @cortinico

Comment (muhammadashfaq) [reply docs/image.md]: @cortinico added as a paragraph. Please take a look. Thanks

File: docs/image.md [hunk 2]
```diff
@@ -419,7 +419,9 @@ The image source (either a remote URL or a local file resource).
 
 This prop can also contain several remote URLs, specified together with their width and height and potentially with scale/other URI arguments. The native side will then choose the best `uri` to display based on the measured size of the image container. A `cache` property can be added to control how networked request interacts with the local cache. (For more information see [Cache Control for Images](images#cache-control-ios-only)).
 
-The currently supported formats are `png`, `jpg`, `jpeg`, `bmp`, `gif`, `webp`, `psd` (iOS only). In addition, iOS supports several RAW image formats. Refer to Apple's documentation for the current list of supported camera models (for iOS 12, see https://support.apple.com/en-ca/HT208967).
+The currently supported formats are `png`, `jpg`, `jpeg`, `bmp`, `gif`,`webp`, `psd` (iOS only).Refer to Apple's documentation for the current list of supported camera models (for iOS 12, see https://support.apple.com/en-ca/HT208967).
```
Comment (cortinico) [inline docs/image.md:422]:
```suggestion
The currently supported formats are `png`, `jpg`, `jpeg`, `bmp`, `gif`, `webp`, `psd` (iOS only). In addition, iOS supports several RAW image formats. Refer to Apple's documentation for the current list of supported camera models (for iOS 12, see https://support.apple.com/en-ca/HT208967).
```

File: docs/image.md [hunk 3]
```diff
@@ -419,7 +419,9 @@ The image source (either a remote URL or a local file resource).
...
+
+In addition, iOS supports several RAW image formats. `webp` format is supported on iOS only when bundled with the JavaScript code.
```
Comment (cortinico) [inline docs/image.md:424]:
```suggestion
Please note that the `webp` format is supported on iOS **only** when bundled with the JavaScript code.
```

**Pull request merged**